invenio_rdm_migrator/extract/kafka.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Kafka extract class."""
+
+"""
+# TODO: remove `schema` from debezium message
+
+# TODO: remove from debezium config the following tables:
+- pidstore_recid (incremental index)
+- transaction
+- record_metadata_version
+- record_buckets
+- pidrelations (all models)
+- sipstore (all models)
+
+# TODO: deactivate on Zenodo the integrity check task
+
+# TODO: Implement an extract class that would consume messages from a topic
+
+class KafkaExtract(Extract):
```

Comment:
Skeleton for a kafka consumer this would serve both for the "grouping" as well as the actual "action" (transaction streaming). 

The subsequent transformer expects the following format:

```
- tx_id: actual transaction id of the current group
- tx_group_id: ad-hoc, natural language id which can be omitted, but would be helpful to debug. e.g. "draft-create", "file-upload"
- operations : a list of of dictionaries with "op" (create, update, delete) and "data" (the actual replication message from where to extract the  `after` payload).
```